SensorApplication.java

```diff
@@ -0,0 +1,69 @@
+package sensor;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.GridLayout;
+
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.border.TitledBorder;
+
+public class SensorApplication extends JFrame {
+	
+	public SensorApplication() {
+		setTitle("Sensor Tracker");
+		setLayout(new GridLayout(3,1));
+		
+        Sensor[] sensors = {
+        		new PressureSensorAdapter(new PressureSensor()),
+                new RadiationSensorAdapter(new RadiationSensor()),
+                new TemperatureSensorAdapter(new TemperatureSensor())
+        };
+        
+        for (Sensor sensor : sensors) {
+    		JPanel sensorPnl = new JPanel();
+    		sensorPnl.setBorder(new TitledBorder(sensor.getName()));
+    		double value = sensor.getValue();
+    		String report = sensor.getReport();
+    		JTextArea colorTextArea = this.getColorComponent(report);
+    		JLabel reportValueLbl = new JLabel(report + " --> " + String.valueOf(value));    		
+    		sensorPnl.add(colorTextArea);
+    		sensorPnl.add(reportValueLbl);
+    		add(sensorPnl);
+        }
+		
+		setPreferredSize(new Dimension(600,600));
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setVisible(true);
+		pack();
+	}
+	

```

Comment:
The getColorComponent method contains logic that should probably be encapsulated in the adapter classes. In general GUI(SensorApplication) should have little to no business logic about other classes. The GUI should be primarily concerned about creating interface components and using APIs to get required parameters. If the adaptee classes get refactored, the adapter classes should be the only ones affected. As of right now, any changes made to the adaptee classes could affect both the adapter and the GUI. For example, if we wanted to refactor the adaptee classes and have them report "EXCELLENT", "OKAY", and "BAD" instead of "OK", "CRITICAL", and "DANGER" we would have to modify the logic in the GUI. The more business logic the GUI contains, the higher the coupling.